typed-racket-lib/typed-racket/infer/infer-unit.rkt

```diff
@@ -720,8 +723,15 @@
           [((CustodianBox: t) (Evt: t*)) (cg S t*)]
           [((Channel: t) (Evt: t*)) (cg t t*)]
           [((Async-Channel: t) (Evt: t*)) (cg t t*)]
-          ;; we assume all HTs are mutable at the moment
-          [((Hashtable: s1 s2) (Hashtable: t1 t2))
+          [((Immutable-Hashtable: s1 s2)
+            (or (Hashtable: t1 t2)
+                (Immutable-Hashtable: t1 t2)))
+           ;; for immutable hash tables, covariant
+           (% cset-meet (cg s1 t1) (cg s2 t2))]
```

Comment:
I think it's fine because the value is invariant. You're safe to treat an (IH A B) as an (H A' B') because any write to the (H ...) is a runtime error.

If we remove this case then `for/hash` needs to return a `HashTable` instead of an `Immutable-HashTable`, otherwise unit tests fail. E.g.
```
#lang typed/racket/base

(for/hash : (HashTable Integer Integer)
          ([x (in-list '(1 2 3))])
  (values x x))
```